src/Repository.ts

```diff
@@ -265,9 +267,19 @@ export class MongoRepository<DOC, DTO = DOC> {
   async deleteMany(conditions: any): Promise<DeleteWriteOpResultObject> {
     const collection = await this.collection;
 
-    await this.invokeEvents(PRE_KEY, [RepoOperation.delete, RepoOperation.deleteMany], conditions);
+    const documents = (await collection.find(conditions).toArray()).map(document => {
+      return this.toggleId(document, false);
+    });
+
+    for (const document of documents) {
+      await this.invokeEvents(PRE_KEY, [RepoOperation.delete, RepoOperation.deleteMany], null, document);
```

Comment:
Much like find, the `newDocument` is the document retrieved, you would only use `originalDocument` if there was changes (and should not be passed in, in this case)

Reply:
the function arguments naming should probably be revisited to make it more clear